entry/src/entry.rs

```diff
@@ -524,46 +524,48 @@ fn start_verify_transactions_gpu(
         });
     }
 
-    let mut packet_batches = entry_txs
-        .par_iter()
-        .chunks(PACKETS_PER_BATCH)
-        .map(|slice| {
-            let vec_size = slice.len();
-            let mut packet_batch = PacketBatch::new_with_recycler(
-                &verify_recyclers.packet_recycler,
-                vec_size,
-                "entry-sig-verify",
-            );
-            // We use set_len here instead of resize(vec_size, Packet::default()), to save
-            // memory bandwidth and avoid writing a large amount of data that will be overwritten
-            // soon afterwards. As well, Packet::default() actually leaves the packet data
-            // uninitialized, so the initialization would simply write junk into
-            // the vector anyway.
-            unsafe {
-                packet_batch.set_len(vec_size);
-            }
-            let entry_tx_iter = slice
-                .into_par_iter()
-                .map(|tx| tx.to_versioned_transaction());
-
-            let res = packet_batch
-                .par_iter_mut()
-                .zip(entry_tx_iter)
-                .all(|(packet, tx)| {
-                    *packet.meta_mut() = Meta::default();
-                    Packet::populate_packet(packet, None, &tx).is_ok()
-                });
-            if res {
-                Ok(packet_batch)
-            } else {
-                Err(TransactionError::SanitizeFailure)
-            }
-        })
-        .collect::<Result<Vec<_>>>()?;
+    let packet_batches = thread_pool.install(|| {
```

Comment:
My understanding is `thread_pool` defaults to num_cpu number of threads. I believe the global rayon pool is 1/2 num_cpu. I could be wrong here.. this is from memory.

But if I have that right, it might be good to understand if there's any performance weirdness that comes from trying to soak up all the CPUs to do this work.

Reply:
The thread pool getting passed in here is this one (we use the same pool for replay + entry verification):
https://github.com/anza-xyz/agave/blob/7640c25509b0aa299b7e3238208c1d01035e7898/core/src/replay_stage.rs#L644-L648
and you're correct that we'll default to `num_logical_cpu`:
https://github.com/anza-xyz/agave/blob/7640c25509b0aa299b7e3238208c1d01035e7898/validator/src/cli/thread_args.rs#L157

Here is that definition, you can see a that we do `(num_cpus::get() / 2 ) * 2` in that file

https://github.com/anza-xyz/agave/blob/7640c25509b0aa299b7e3238208c1d01035e7898/rayon-threadlimit/src/lib.rs#L23-L25
The global Rayon threadpool also has `num_logical_cpu` threads as well per [here](https://docs.rs/rayon/latest/rayon/struct.ThreadPoolBuilder.html#method.num_threads) since we don't set the number of threads at initialization:
https://github.com/anza-xyz/agave/blob/7640c25509b0aa299b7e3238208c1d01035e7898/core/src/validator.rs#L589-L595